From 03b41921dcf395be4ad5345dbd2c96338b192a58 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 27 Oct 2026 13:16:54 +0100
Subject: [PATCH 10/N] Cygwin: strace: keep argument scanning const-correct

Building `strace` with a recent GCC version fails on a const-discarding pointer
conversion in `make_command_line()`. There is no need to discard the `const`,
though: the logic scanning the arguments do not modify the data, and the
command-line is assembled into a separate buffer.

Assisted-by: GPT-6.1 Sol
Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/utils/mingw/strace.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/utils/mingw/strace.cc b/winsup/utils/mingw/strace.cc
index c220643..5b6eadb 100644
--- a/winsup/utils/mingw/strace.cc
+++ b/winsup/utils/mingw/strace.cc
@@ -228,7 +228,7 @@ make_command_line (linebuf & one_line, char **argv)
 {
   for (; *argv; argv++)
     {
-      char *p = NULL;
+      const char *p = NULL;
       const char *a = *argv;
 
       int len = strlen (a);
